snapshot/restore/local_fs_output.go

```diff
@@ -139,6 +167,16 @@ func fileIsSymlink(stat os.FileInfo) bool {
 	return stat.Mode()&os.ModeSymlink != 0
 }
 
+// SymlinkExists implements restore.Output interface.
+func (o *FilesystemOutput) SymlinkExists(ctx context.Context, relativePath string, e fs.Symlink) bool {
+	st, err := os.Lstat(filepath.Join(o.TargetPath, relativePath))
+	if err != nil {
+		return false
+	}
+
+	return (st.Mode() & os.ModeType) == os.ModeSymlink
```

Comment:
Should this also check whether `o.TargetPath/relativePath` points to the same target as `e` ?

Reply:
the whole point was to avoid reading objects from repository, so we can either always read or be relatively lax here.

Reply:
in hindsight symlink target should have been part of metadata and not stored in an object, perhaps we can fix that...

Reply:
Then, the log message below is not accurate and could be misleading

`log(ctx).Debugf("skipping symlink %v because it already exists and is correct", targetPath)`